src/Octoshift/AwsApi.cs

```diff
@@ -5,24 +5,40 @@
 using Amazon.S3;
 using Amazon.S3.Model;
 using Amazon.S3.Transfer;
+using OctoshiftCLI.Extensions;
 
 namespace OctoshiftCLI;
 
 public class AwsApi : IDisposable
 {
     private const int AUTHORIZATION_TIMEOUT_IN_HOURS = 48;
-    private static readonly RegionEndpoint RegionEndpoint = RegionEndpoint.USEast1;
+    private static readonly RegionEndpoint DefaultRegionEndpoint = RegionEndpoint.USEast1;
 
     private readonly ITransferUtility _transferUtility;
 
 #pragma warning disable CA2000
-    public AwsApi(string awsAccessKey, string awsSecretKey) : this(new TransferUtility(new AmazonS3Client(awsAccessKey, awsSecretKey, RegionEndpoint)))
+    public AwsApi(string awsAccessKey, string awsSecretKey, string awsRegion = null, string awsSessionToken = null)
+        : this(new TransferUtility(BuildAmazonS3Client(awsAccessKey, awsSecretKey, awsRegion, awsSessionToken)))
 #pragma warning restore CA2000
     {
     }
 
     internal AwsApi(ITransferUtility transferUtility) => _transferUtility = transferUtility;
 
+    private static AmazonS3Client BuildAmazonS3Client(string awsAccessKey, string awsSecretKey, string awsRegion, string awsSessionToken)
+    {
+        var regionEndpoint = DefaultRegionEndpoint;
+        if (awsRegion.HasValue())
+        {
+            regionEndpoint = RegionEndpoint.GetBySystemName(awsRegion);
```

Comment:
Will this throw an exception if an invalid region name is specified? I just want to make sure I understand our failure modes!

Reply:
@timrogers I am not sure how I missed this comment, but this is a very valid question. I was naturally under the impression that `GetBySystemName()` would throw if the region string was invalid but interestingly that's not the case! `GetBySystemName()` always succeeds even for an invalid region name! Not sure why they don't validate it! The migration fails eventually but we should def. validate and throw if the region is invalid. Fail fast, fail early 😄 
I am going to put together a quick fix for this. Thanks for pointing this out 🙏

Reply:
PR for the fix --> https://github.com/github/gh-gei/pull/872